modules/lemmaBidAdapter.js

```diff
@@ -0,0 +1,406 @@
+        domain: appParams.domain,
+        cat: appParams.categories,
+        pagecat: appParams.page_category
+      };
+    }
+  }
+  return null;
+}
+
+function _getDeviceObject(request) {
+  var params = request && request.params ? request.params : null;
+  if (cfn(params) != '') {
+    return {
+      dnt: utils.getDNT() ? 1 : 0,
+      ua: navigator.userAgent,
+      language: (navigator.language || navigator.browserLanguage || navigator.userLanguage || navigator.systemLanguage),
+      w: (window.screen.width || window.innerWidth),
+      h: (window.screen.height || window.innerHeigh),
+      geo: {
+        country: params.country,
+        lat: params.latitude,
+        lon: params.longitude,
+        region: params.region,
+        city: params.city,
+        zip: params.zip
+      },
+      ip: params.ip,
+      devicetype: params.device_type,
+      ifa: params.ifa,
+    };
+  }
+  return null;
+}
+
+function setOtherParams(request, ortbRequest) {
+  var params = request && request.params ? request.params : null;
+  if (request && request.gdprConsent) {
+    ortbRequest.regs = { ext: { gdpr: request.gdprConsent.gdprApplies ? 1 : 0 } };
+    ortbRequest.user = { ext: { consent: request.gdprConsent.consentString } };
+  }
+  if (cfn(params) != '') {
+    ortbRequest.tmax = params.tmax;
+    ortbRequest.bcat = params.bcat;
+  }
+}
+
+function _getSizes(request) {
+  var size = cfn(cfn(request).sizes)[0];
+  if (utils.isArray(size) && cfn(size).length > 0) {
+    return size;
+  }
+  return null;
+}
+
+function _getBannerRequest(bid) {
+  var bObj;
+  var adFormat = [];
+  if (cfn(bid.mediaType) === 'banner' || utils.deepAccess(bid, 'mediaTypes.banner')) {
+    var params = cfn(bid) != '' ? bid.params : null;
+    var bannerData = cfn(params.banner);
+    var sizes = _getSizes(bid);
+    if (cfn(sizes).length == 0) {
+      sizes = cfn(bid.mediaTypes.banner.sizes[0]);
+    }
+    if (cfn(sizes).length > 0) {
+      bObj = {};
+      bObj.w = cfn(sizes[0]);
+      bObj.h = cfn(sizes[1]);
+      bObj.pos = 0;
+      if (cfn(bannerData) != '') {
+        bObj = utils.deepClone(bannerData);
+      }
+      sizes = cfn(bid.mediaTypes.banner.sizes);
+      if (cfn(sizes).length > 0) {
+        adFormat = [];
+        sizes.forEach(function(size) {
+          if (size.length > 1) {
+            adFormat.push({ w: size[0], h: size[1] });
+          }
+        });
+        if (adFormat.length > 0) {
+          bObj.format = adFormat;
+        }
+      }
+    } else {
+      utils.logWarn(LOG_WARN_PREFIX + 'Error: mediaTypes.banner.sizes missing for adunit: ' + bid.params.adunitId);
+    }
+  }
+  return bObj;
+}
+
+function _getVideoRequest(bid) {
+  var vObj;
+  if (cfn(bid.mediaType) === 'video' || utils.deepAccess(bid, 'mediaTypes.video')) {
+    var params = cfn(bid) != '' ? bid.params : null;
+    var sizes = _getSizes(bid);
+    if (cfn(sizes).length == 0) {
+      sizes = cfn(bid.mediaTypes.video.playerSize);
```

Comment:
Is this condition reachable? It looks like `_getSizes` will return null if length is 0.

Reply:
If the request object does not contain sizes parameter, then this condition reached. I have updated condition.